lib/segment/src/index/plain_payload_index.rs

```diff
@@ -278,6 +279,7 @@ impl VectorIndex for PlainIndex {
         top: usize,
         params: Option<&SearchParams>,
         query_context: &VectorQueryContext,
+        _hardware_counter: &HardwareCounterCell,
```

Comment:
WDYT about putting `HardwareCounterCell` into `SegmentQueryContext`?

It is also per-segment, we should not have problems with sync/send, and it is already propagated everywhere (via VectorQueryContext). 

Overall it should have less impact on the code changes in tests